cactus/plugin/builtin/pagecontext.py

```diff
@@ -0,0 +1,120 @@
+#coding:utf-8
+import logging
+import re
+
+
+logger = logging.getLogger(__name__)
+
+
+class PageContextPlugin(object):
```

Comment:
The "contract" used to be that context would be loaded before any plugin runs. Can you ensure that this continues to be the case here? 

Plugins are sorted as per their `ORDER` property. You can probably design an `ORDER` property that sorts lower than everything (and therefore always loads first).